charts/prometheus-elasticsearch-exporter/templates/podmonitor.yaml

```diff
@@ -1,7 +1,11 @@
 {{- if and .Values.podMonitor.enabled .Values.serviceMonitor.enabled }}
 {{- fail "Either .Values.podMonitor.enabled or .Values.serviceMonitor.enabled can be enabled at a time, but not both." }}
 {{- else if .Values.podMonitor.enabled }}
+{{- if .Values.azuremanagedprometheus.use_azuremanagedprometheus }}
+apiVersion: azmonitoring.coreos.com/v1
+{{- else }}
 apiVersion: monitoring.coreos.com/v1
```

Comment:
If schema for this objects is really the same(i have no idea), it might be easier to just template `apiVersion` value and keep `monitoring.coreos.com/v1` default for backwards compatibility as we might have more of these in the future (GCP has a similar config)

Reply:
I would also potentially add a check with something like`{{- if .Capabilities.APIVersions.Has "azmonitoring.coreos.com/v1" }}` for extra validation

Reply:
Thanks @DrFaust92 for the feedback. The schema is the same. I have addressed your feedback. Please let me know if this looks good.